server/test/routes/index.test.ts

```diff
@@ -161,6 +157,117 @@ describe('Test GET /v1/user/:userId/courses', () => {
+    let input: object = {
+      courseCode: 'ELEC-A7200',
+      department: {
+        fi: 'Sähkötekniikan korkeakoulu',
+        en: 'School of Electrical Engineering',
+        sv: 'Högskolan för elektroteknik'
+      },
+      name: {
+        fi: 'Signaalit ja järjestelmät',
+        en: 'Signals and Systems',
+        sv: ''
+      }
+    };
+    let res: supertest.Response = await request.post('/v1/courses').send(input);
+    expect(res.statusCode).toBe(200);
+    expect(res.body.success).toBe(true);
+    expect(res.body.course.courseCode).toBe('ELEC-A7200');
+    expect(res.body.course.name).toStrictEqual({
+      'fi': 'Signaalit ja järjestelmät',
+      'en': 'Signals and Systems',
+      'sv': ''
+    });
+    expect(res.body.course.department).toStrictEqual({
+      'fi': 'Sähkötekniikan korkeakoulu',
+      'en': 'School of Electrical Engineering',
+      'sv': 'Högskolan för elektroteknik'
+    });
+
+    input = {
+      courseCode: 'ELEC-A7200',
+      department: {
+        fi: 'Sähkötekniikan korkeakoulu',
+        en: 'School of Electrical Engineering',
+      },
+      name: {
+        fi: 'Signaalit ja järjestelmät',
+        en: 'Signals and Systems',
+      }
+    };
+    res = await request.post('/v1/courses').send(input);
+    expect(res.statusCode).toBe(200);
+    expect(res.body.success).toBe(true);
+    expect(res.body.course.courseCode).toBe('ELEC-A7200');
+    expect(res.body.course.name).toStrictEqual({
+      'fi': 'Signaalit ja järjestelmät',
+      'en': 'Signals and Systems',
+      'sv': ''
+    });
+    expect(res.body.course.department).toStrictEqual({
+      'fi': 'Sähkötekniikan korkeakoulu',
+      'en': 'School of Electrical Engineering',
+      'sv': ''
+    });
+  });
+
+  it('should respond with validation errors, if required fields are undefined', async () => {
+    const input: object = {};
+    const res: supertest.Response = await request.post('/v1/courses').send(input);
+    expect(res.statusCode).toBe(400);
+    expect(res.body.success).toBe(false);
+    expect(res.body.errors).toContain('courseCode is a required field');
+    expect(res.body.errors).toContain('department is a required field');
+    expect(res.body.errors).toContain('name is a required field');
+  });
+
+  it('should respond with validation errors, if the object for a localized string is empty', async () => {
+    const input: object = {
+      courseCode: 'ELEC-A7200',
+      department: {},
+      name: {}
+    };
+    const res: supertest.Response = await request.post('/v1/courses').send(input);
+    expect(res.statusCode).toBe(400);
+    expect(res.body.success).toBe(false);
+    expect(res.body.errors).toContain('department must contain at least one translation');
+    expect(res.body.errors).toContain('name must contain at least one translation');
+  });
+
+  it('should respond with validation errors, if unknown translations are present in a localized string', async () => {
```

Comment:
These localized string tests are more general than just this route, and they should be tested separately.

Reply:
I could probably create test/controllers/course.test.ts module where to create unit tests for functions in that module.

Reply:
We should also discuss more about our testing strategy in next sprint. And also refactoring the code so that commonly reusable operations would be separated into their own functions (maybe even in separate modules?) and tested then separately as well.